Source/Machine.Fakes.FakeItEasy/FakeItEasyExpressionRewriter.cs

```diff
@@ -66,27 +68,23 @@ static Expression RewriteIsNullMember(MemberExpression node)
             var argumentType = node.Member.DeclaringType.GetFirstTypeArgument();
             var thatAccess = GetThatAccess(argumentType);
 
-            return Expression.Call(
-                typeof(ArgumentConstraintManagerExtensions),
-                "IsNull",
-                new[] { argumentType },
-                thatAccess);
+            var method = GetIsNullMethod(argumentType);
+
+            return Expression.Call(method, thatAccess);
         }
 
         static Expression RewriteIsNotNullMember(MemberExpression node)
         {
             var argumentType = node.Member.DeclaringType.GetFirstTypeArgument();
             var thatAccess = GetThatAccess(argumentType);
 
-            var notAccess = typeof(IArgumentConstraintManager<>)
+            var notAccess = typeof(INegatableArgumentConstraintManager<>)
```

Comment:
An alternative would be to right at the [IsNotNull method](http://fakeiteasy.readthedocs.io/en/stable/argument-constraints/#more-convenience-matchers), which is relatively new. But since you need a way to find `IsNull` anyhow, there's no harm in this approach.